Cesium3DTilesSelection/src/Tile.cpp

```diff
@@ -257,62 +259,88 @@ void Tile::loadContent() {
     std::unique_ptr<TileContentLoadResult> pContent;
     void* pRendererResources;
   };
-  TileContentLoadInput loadInput(tileset.getExternals().pLogger, *this);
 
   const CesiumGeometry::Axis gltfUpAxis = tileset.getGltfUpAxis();
   std::move(maybeRequestFuture.value())
-      .thenInWorkerThread(
-          [loadInput = std::move(loadInput),
+      // TODO: for some reason, this first lambda completely breaks if we make
+      // it mutable
+      .thenInWorkerThread([this,
+                           &asyncSystem = tileset.getAsyncSystem(),
+                           &pLogger_ = tileset.getExternals().pLogger,
+                           &pAssetAccessor_ =
+                               tileset.getExternals().pAssetAccessor](
```

Comment:
Since the lambda can be mutable now I can pass the shared_ptr by value now. But as for the "this" pointer, I thought a Tile was guaranteed to not be deleted when its content was loading? Or is that not always the case?

Reply:
That's true, the Tile won't be deleted while the content is loading. But passing `this` is still dangerous because the Tile is not thread safe and so _using_ the `this` is unsafe. And it's especially easy to do that accidentally because `this` is implicit. So best thing is to avoid passing non-thread-safe objects to the worker thread lambda in general, but if you need to you should at least avoid naming them `this`. e.g. `[pTile = this, ...]` is an improvement.